test/modules/aws-ecs-integration.ts

```diff
@@ -97,43 +122,35 @@ describe('ECS Integration Testing', () => {
     `, (res: any[]) => expect(res.length).toBe(1)));
 
     it('adds a new container definition', query(`
-      INSERT INTO aws_container_definition ("name", docker_image, tag, essential, memory_reservation, host_port, container_port, protocol, env_variables, task_definition_id, log_group_id)
-	    VALUES('${containerName}', '${image}', '${imageTag}', ${containerEssential}, ${containerMemoryReservation}, ${hostPort}, ${containerPort}, '${protocol}', '{ "test": 2}', (select id from aws_task_definition where family = '${tdFamily}' and status is null limit 1), (select id from log_group where log_group_name = '${logGroupName} limit 1'));
+      BEGIN;
+        INSERT INTO aws_container_definition ("name", image, essential, memory_reservation, host_port, container_port, protocol, env_variables, task_definition_id, log_group_id)
+        VALUES('${containerName}', '${image}', ${containerEssential}, ${containerMemoryReservation}, ${hostPort}, ${containerPort}, '${protocol}', '{ "test": 2}', (select id from aws_task_definition where family = '${tdFamily}' and status is null limit 1), (select id from log_group where log_group_name = '${logGroupName} limit 1'));
+        INSERT INTO aws_container_definition ("name", image, tag, essential, memory_reservation, host_port, container_port, protocol, env_variables, task_definition_id, log_group_id)
+        VALUES('${containerNameTag}', '${image}', '${imageTag}', false, ${containerMemoryReservation}, ${hostPort + 1}, ${containerPort + 1}, '${protocol}', '{ "test": 2}', (select id from aws_task_definition where family = '${tdFamily}' and status is null limit 1), (select id from log_group where log_group_name = '${logGroupName} limit 1'));
+        INSERT INTO aws_container_definition ("name", image, digest, essential, memory_reservation, host_port, container_port, protocol, env_variables, task_definition_id, log_group_id)
+        VALUES('${containerNameDigest}', '${image}', '${imageDigest}', false, ${containerMemoryReservation}, ${hostPort + 2}, ${containerPort + 2}, '${protocol}', '{ "test": 2}', (select id from aws_task_definition where family = '${tdFamily}' and status is null limit 1), (select id from log_group where log_group_name = '${logGroupName} limit 1'));
+      COMMIT;
     `));
 
     it('check container definition insertion', query(`
       SELECT *
       FROM aws_container_definition
-      WHERE name = '${containerName}' AND docker_image = '${image}' AND tag = '${imageTag}';
+      WHERE name = '${containerName}' AND image = '${image}';
```

Comment:
this is the one without the type safety and was causing the error, right?

Reply:
hmm I do not follow the question. The issue was that when no repository found it was thawing an [error](https://github.com/iasql/iasql-engine/pull/509/files#diff-5ec517f3f47bf40ef7dea8ec93f031da5633f9ed205c9322c8491046a4752b4cL263). Now it just try to get the related ECR but if it is not found, it just save the value in the `image` field as a `string`.

Reply:
I mean the error you found in the production logs after the call earlier this week

Reply:
oh, it is this:
> The issue was that when no repository found it was throwing an [error](https://github.com/iasql/iasql-engine/pull/509/files#diff-5ec517f3f47bf40ef7dea8ec93f031da5633f9ed205c9322c8491046a4752b4cL263).